core/sawtooth/client.py

```diff
@@ -54,8 +54,8 @@ def _sign_message_with_transaction(transaction, message_type, key):
     of a sha256 hexdigest.
     """
     transaction['Nonce'] = time.time()
```

Comment:
I think I would just leave this as-is.  It is probably broken in so many other ways :)